### PR TITLE
Upgrade Prettier version, fix comment discovery

### PR DESCRIPTION
**Pull request by FuegoFro**
Bump the dependency on the main Prettier repo to the latest commit. This allows us to specify additional options (such as `pythonVersion`). Note that the Python version defaults to Python3, which I believe matches the behavior from before.

The plugin API in the newer version requires plugins to provide a `locStart` and `locEnd` function. From what I've seen, this is to allow customization, which is something we want and will help with things like better comment rendering. However, I'm not totally sure what customizations we want/need in these functions, so for now this just copies in the old versions of those functions. This shouldn't actually be a functional change from before the upgrade, and still allows us to modify/customize that functionality in the future as needed.

I also noticed a bug in how comments are detected within Python. There was a check that hard-coded the `COMMENT` token type as `57`, but it's actually different in different versions of Python. This fixes that check to use the proper value for the given version of Python and updates a test that mistakenly did not have comments in the expected output.

### CODE REVIEW

**Comment by taion** [comment]
@FuegoFro See also https://github.com/prettier/prettier-python/pull/15 which deals with a lot of those issues and adds checks that things are consistent between Python 2 and 3.

**Comment by FuegoFro** [comment]
Oh nice! I like a lot of the Python stuff done in that diff! It seems like the overlap with this (using the `tokenizer.COMMENT` constant and modifying the `pythonVersions`) are small enough that these two can be considered mostly independent? Specifically the `tokenizer.COMMENT` change is just one line, and the `options` object I added can be removed or modified and necessary once your more complete Python2/3 handling lands. Given that it seems that we wouldn't necessarily need to block this PR on that one. Does that seem like a reasonable/accurate description?

**Comment by taion** [comment]
Yeah, that sounds totally reasonable. I haven't had much time to think about much of this – just wanted to point you at another attempt to handle some of the same issues.

**Comment by FuegoFro** [comment]
Looks like CI might have had a flake. Closing and re-opening to try to force Travis to rebuild. (Is there a better way to do that?)

File: src/index.js [hunk 1]
```diff
@@ -1,5 +1,6 @@
 "use strict";
 
+const utilShared = require("prettier").util;
 const util = require("./_util-from-prettier");
```
Comment (azz) [inline src/index.js]: Can we delete this file now?

Comment (FuegoFro) [reply src/index.js]: Ah yeah, good call! It's gone!

File: src/index.js [hunk 2]
```diff
@@ -47,7 +133,6 @@ function clean(ast, newObj) {
 const printers = {
   python: {
     print,
-    hasPrettierIgnore: util.hasIgnoreComment,
```
Comment (FuegoFro) [inline src/index.js]: Note that this wasn't actually being passed in, as far as I can tell (the value was `undefined`, even on master).

**Comment by patrick91** [comment]
@FuegoFro what's the state of this?

**Comment by FuegoFro** [comment]
Just rebased and fixed up conflicts and failing tests!

**Pull request merged**